src/module.ts

```diff
@@ -0,0 +1,88 @@
+import { resolve } from 'path'
+import { fileURLToPath } from 'url';
+import { defineNuxtModule, addPlugin, isNuxt2 } from '@nuxt/kit'
+import defu from 'defu'
+
+import logger from './logger'
+
+export interface ModuleOptions {
+	id?: string,
+	dev?: boolean,
+	metrikaUrl?: string,
+	accurateTrackBounce?: boolean | number,
+	childIframe?: boolean,
+	clickmap?: boolean,
+	defer?: boolean,
+	ecommerce?: boolean | string | [],
+	params?: object | [],
+	useCDN?: boolean,
+	userParams?: object,
+	trackHash?: boolean,
+	trackLinks?: boolean,
+	trustedDomains?: [],
+	type?: number,
+	webvisor?: boolean,
+	triggerEvent?: boolean,
+}
+
+const CONFIG_KEY = 'yandexMetrika'
+
+declare module '@nuxt/schema' {
+	interface PublicRuntimeConfig {
+		[CONFIG_KEY]?: ModuleOptions
+	}
+}
+
+export default defineNuxtModule<ModuleOptions>({
+	meta: {
+		name: '@nuxtjs/yandex-metrika',
+		configKey: CONFIG_KEY
+	},
```

Comment:
we could also add compatibility information here - e.g. does it require bridge, does it work with nuxt 2, etc.